sqlmesh/core/engine_adapter/mssql.py

```diff
@@ -206,3 +209,80 @@ def _truncate_table(self, table_name: TableName) -> str:
     def _to_sql(self, expression: exp.Expression, quote: bool = True, **kwargs: t.Any) -> str:
         sql = super()._to_sql(expression, quote=quote, **kwargs)
         return f"{sql};"
+
+    @overload
+    def execute_and_fetch(
+        self,
+        expression: t.Union[str, exp.Expression],
+        fetch_one: Literal[False] = ...,
+        ignore_unsupported_errors: bool = False,
+        quote_identifiers: bool = True,
+        **kwargs: t.Any,
+    ) -> t.List[t.Tuple]:
+        ...
+
+    @overload
+    def execute_and_fetch(
+        self,
+        expression: t.Union[str, exp.Expression],
+        fetch_one: Literal[True],
+        ignore_unsupported_errors: bool = False,
+        quote_identifiers: bool = True,
+        **kwargs: t.Any,
+    ) -> t.Tuple:
+        ...
```

Comment:
I think we may be able to reduce the visual noise here by doing something similar to this: https://github.com/tobymao/sqlglot/blob/main/sqlglot/__init__.py#L89